src/backend/src/controllers/change-requests.controllers.ts

```diff
@@ -1,509 +1,104 @@
-      },
-      include: {
-        workPackage: true,
-        project: true
-      }
-    });
-    if (!wbs) {
-      return res.status(404).json({ message: `WBS element with id #${foundCR.wbsElementId} not found` });
-    }
-    const { workPackage, project } = wbs;
-
-    if (!workPackage && project) {
-      const newBudget = project.budget + foundPs.budgetImpact;
-      const change = {
-        changeRequestId: crId,
-        implementerId: reviewerId,
-        detail: buildChangeDetail('Budget', String(project.budget), String(newBudget))
-      };
-      await prisma.project.update({
-        where: { projectId: project.projectId },
-        data: {
-          budget: newBudget,
-          wbsElement: {
-            update: {
-              changes: {
-                create: change
-              }
-            }
-          }
-        }
-      });
-    } else if (workPackage) {
-      const wpProj = await prisma.project.findUnique({
-        where: { projectId: workPackage.projectId }
-      });
-      if (!wpProj) {
-        return res.status(404).json({ message: 'Work package project not found' });
-      }
-      const newBudget = wpProj.budget + foundPs.budgetImpact;
-      const updatedDuration = workPackage.duration + foundPs.timelineImpact;
-
-      const changes = [
-        {
-          changeRequestId: crId,
-          implementerId: reviewerId,
-          detail: buildChangeDetail('Budget', String(wpProj.budget), String(newBudget))
-        },
-        {
-          changeRequestId: crId,
-          implementerId: reviewerId,
-          detail: buildChangeDetail('Duration', String(workPackage.duration), String(updatedDuration))
-        }
-      ];
-      await prisma.project.update({
-        where: { projectId: workPackage.projectId },
-        data: {
-          budget: newBudget,
-          workPackages: {
-            update: {
-              where: { workPackageId: workPackage.workPackageId },
-              data: {
-                duration: updatedDuration,
-                wbsElement: {
-                  update: {
-                    changes: {
-                      create: changes[1]
-                    }
-                  }
-                }
-              }
-            }
-          },
-          wbsElement: {
-            update: {
-              changes: {
-                create: changes[0]
-              }
-            }
-          }
-        }
-      });
+import { Request, Response, NextFunction } from 'express';
+import ChangeRequestService from '../services/change-request.services';
+
+export default class CRController {
+  // Fetch the specific change request by its integer ID
+  static async getChangeRequestByID(req: Request, res: Response, next: NextFunction) {
+    try {
+      const crId: number = parseInt(req.params.crId);
+      const cr = await ChangeRequestService.getChangeRequestByID(crId);
+      return res.status(200).json(cr);
+    } catch (error: unknown) {
```

Comment:
Is there anything better than `unknown` that can be used instead?

Reply:
I just used unknown cause thats what is used in the risks file

Reply:
no this is how typescript works unfortunately

Reply:
the error handler checks what type of error it is though